crates/stackable-operator/src/commons/listener.rs

```diff
@@ -68,6 +68,34 @@ pub enum ServiceType {
     ClusterIP,
 }
 
+impl ServiceType {
+    pub fn to_kubernetes_literal(&self) -> String {
+        match self {
+            ServiceType::NodePort => "NodePort".to_owned(),
+            ServiceType::LoadBalancer => "LoadBalancer".to_owned(),
+            ServiceType::ClusterIP => "ClusterIP".to_owned(),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, Default, JsonSchema, PartialEq, Eq)]
+pub enum TrafficPolicy {
+    /// Obscures the client source IP and may cause a second hop to another node, but allows Kubernetes to spread the load between all nodes.
+    #[default]
+    Cluster,
+    /// Preserves the client source IP and avoid a second hop for LoadBalancer and NodePort type Services, but makes clients responsible for spreading the load.
+    Local,
+}
+
+impl TrafficPolicy {
+    pub fn to_kubernetes_literal(&self) -> String {
+        match self {
+            TrafficPolicy::Cluster => "Cluster".to_string(),
+            TrafficPolicy::Local => "Local".to_string(),
+        }
+    }
+}
+
```

Comment:
I think we can probably just impl `std::string::ToString`? Or is there something special about the Kubernetes literal (if there was, I'd assume we'd be implementing a trait anyway)?

Reply:
Thinking about this, let's just name it `KubernetesTrafficPolicy` to make clear it's naming is coming from k8s and use strum::Display

Reply:
1e8571c